noisy-simulator/src/state_vector_simulator/tests.rs

```diff
@@ -0,0 +1,62 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+use super::StateVectorSimulator;
+use crate::tests;
+
+#[test]
+fn measure_0() {
+    tests::measure_0::<StateVectorSimulator>();
+}
+
+#[test]
+fn measure_1() {
+    tests::measure_1::<StateVectorSimulator>();
+}
+
+#[test]
+fn bell_pair_sampling() {
+    // We perform the test 100 times because of the probabilistic nature of the MZ measurement.
+    for seed in 0..100 {
```

Comment:
From what I can tell, the measurement should always yield the same result for the same seed, and in this case in particular the test is only asserting that both qubits are measured in the same state. Given this, what is the reasoning behind testing with different seeds?

Reply:
Since the MZ has randomness associated with it, after you measure the first qubit you will either see a 0 or a 1 with 50% probability. After that the second measurement should yield the same result as the first one.

If you only perform one test with a single seed, you will either be checking that both are 0 or 1, but not both cases. The reasoning behind multiple seeds is to test both cases.

We actually don't need 100 shots to make sure we are checking both cases. With only 20 shots we will test both scenarios with 99.999905% probability. So, I will lower that number to 20.